server-common/src/main/java/org/apache/kafka/server/common/ShareVersion.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.server.common;
+
+import java.util.Map;
+
+public enum ShareVersion implements FeatureVersion {
+
+    // Version 0 does not enable share groups.
+    SV_0(0, MetadataVersion.MINIMUM_VERSION, Map.of()),
+
+    // Version 1 enables share groups (KIP-932).
+    // This is a preview in 4.1, and production-ready in 4.2.
+    SV_1(1, MetadataVersion.IBP_4_2_IV0, Map.of(MetadataVersion.FEATURE_NAME, MetadataVersion.IBP_4_1_IV1.featureLevel()));
```

Comment:
Should we add GroupVersion to its dependency too?

Reply:
This PR is still a draft. I am going to remove the GroupVersion change.

Reply:
But for SV to work, it needs at least some version of GV, right?

Reply:
I would say not now that AK 4.1 contains only the new group coordinator. GV_1 means consumer group protocol is enabled. Share group protocol doesn't rely on that. But @dajac will be the authority on this.